### ISSUE TITLE
Error doesn't get displayed properly when downloading files as anon user

### ISSUE THREAD

**Issue by mschnitzer** [openSUSE/open-build-service#3150]
When I click as an anonymous user the "download file" link (in this example for file `hello_world.spec`), I get redirected back to the file overview page (https://build.opensuse.org/package/show/home:cbruckmayer:oSC17/hello-world) without any error message.

When I copy and paste the link (https://build.opensuse.org/source/home:cbruckmayer:oSC17/hello-world/hello_world.spec?rev=71831c8e605aee4abd8e188854adc6a1) into the address bar of the browser, I get an `anonymous_user(Anonymous user is not allowed here - please login):` error.

a) I don't see a reason why to forbid a user to download files if he/she is not logged in - doesn't make any sense and it's not friendly
b) The error should always be displayed if a) will not be implemented

**Comment by dmarcoux** [comment]
I stumbled on this today and found this issue. I agree that forbidding anonymous users from download files is wrong. As of today, clicking on the `Download File` icon as an anonymous user will display a login popup.

![login](https://user-images.githubusercontent.com/1102934/57088330-59796180-6d02-11e9-8d8f-7d4c1daa8d38.png)

Right now, the `Download File` icon is always displayed for logged-in users. However for anonymous users, it's only when a file is smaller than 4MB.

Downloading a file should always be possible, except for crawlers. We can check the user agent to hide the icon for crawlers. It's easy to fake the user agent though, so we need to have some rate-limiting in place to prevent abuse.

**Comment by hennevogel** [comment]
>  However for anonymous users, it's only when a file is smaller than 4MB.

That does not make much sense if you have to log in afterward right?

**Comment by mschnitzer** [comment]
> Downloading a file should always be possible, except for crawlers. We can check the user agent to
> hide the icon for crawlers. It's easy to fake the user agent though, so we need to have some rate-
> limiting in place to prevent abuse.

The rate limiting is unrelated to the user agent check, but should be there anyway for guests.

We already have `voight_kampff` gem in the codebase, so you could use that one to identify spiders or use `rel="nofollow"`.

**Comment by dmarcoux** [comment]
> > Right now, the Download File icon is always displayed for logged-in users. However for anonymous users, it's only when a file is smaller than 4MB.
> 
> That does not make much sense if you have to log in afterward right?

I agree with you, we need to remove this behavior.

**Comment by dmarcoux** [comment]
> The rate limiting is unrelated to the user agent check, but should be there anyway for guests.

Yes, it's unrelated. Sorry if it wasn't clear in my previous comment.

> We already have `voight_kampff` gem in the codebase, so you could use that one to identify spiders or use `rel="nofollow"`.

Good suggestion to re-use what we already have.
